controllers/topolvm_csi_driver.go

```diff
@@ -0,0 +1,100 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	driverName = "topolvm-csi-driver"
+)
+
+type csiDriver struct{}
+
+// csiDriver unit satisfies resourceManager interface
+var _ resourceManager = csiDriver{}
+
+func (c csiDriver) getName() string {
+	return driverName
+}
+
+//+kubebuilder:rbac:groups=storage.k8s.io,resources=csidrivers,verbs=get;create;delete;watch;list
+
+func (c csiDriver) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+	csiDriverResource := getCSIDriverResource()
+	result, err := cutil.CreateOrUpdate(ctx, r.Client, csiDriverResource, func() error {
+		return nil
```

Comment:
why no `Mutatefn` implementation?  Is the lvmCluster not supposed to be updated once its created?

Reply:
- yes, it's a clusterscoped resource with most of the fields as immutable and mutable fields (name & k8s managed) need not to be mutated